### PR TITLE
Fix bug where after clear chat, new self sent messages are not shown in MessageView

### LINKED ISSUE #3991: message view not always updated
the message view is not always updated; at least not after "clear" in a reproducible way, not sure about other situations.

to reproduce:

- chose "Clear Chat" from the three-dot-menu in the upper right corner
- send a message to that chat from another device (without doing anything on the cleared chat)

-> the message appears in the chatlist, but the chat stays empty.

dc 1.46.1 on mac

https://github.com/deltachat/deltachat-desktop/assets/9800740/908bf7bd-ac1d-48a4-b78d-00f8f99f3af2

### PR DESCRIPTION
**Pull request by farooqkz**
This fixes the bug. But I suspect there are far more bugs here because of usage of `React.memo`. I might be wrong, however so let's wait and see.

Fixes #3991

### CODE REVIEW

**Comment by hpk42** [comment]
There seem to be a lot of unrelated changes in changelog whitespacing.

**Comment by hpk42** [comment]
also which bug is fixed by this?  Please properly reference it with "fixes #..."  so it automatically is referenced bi-directionally?

File: CHANGELOG.md [hunk 1]
```diff
@@ -2701,21 +2847,13 @@ This section is only relevant to contributors.
 
 [unreleased]: https://github.com/deltachat/deltachat-desktop/compare/v1.46.1...HEAD
 
-[1.46.1]: https://github.com/deltachat/deltachat-desktop/compare/v1.46.0...v1.46.1
 
-[1.45.5]: https://github.com/deltachat/deltachat-desktop/compare/v1.45.5...v1.46.0
```
Comment (WofWca) [inline CHANGELOG.md:2706]: Links need to be fixed instead. Plus it'd be better if it was a separate commit, better yet a separate MR.

File: src/renderer/components/message/MessageList.tsx [hunk 2]
```diff
@@ -450,31 +450,31 @@ export type ConversationType = {
...
+    onScroll,
+    messageListItems,
+    messageCache,
+    activeView,
+    messageListRef,
+    chat,
+    loaded,
+    unreadMessageInViewIntersectionObserver,
+    loadMissingMessages,
+  }: MessageListInnerProps) => {
```
Comment (WofWca) [inline src/renderer/components/message/MessageList.tsx:477]: This here is just a refactor, right? I'd put it in a separate commit next time.

File: src/renderer/components/message/MessageList.tsx [hunk 3]
```diff
@@ -644,7 +644,8 @@ export const MessageListInner = React.memo(
       prevProps.messageCache === nextProps.messageCache &&
       prevProps.oldestFetchedMessageIndex ===
         nextProps.oldestFetchedMessageIndex &&
-      prevProps.onScroll === nextProps.onScroll
+      prevProps.onScroll === nextProps.onScroll &&
+      prevProps.messageListItems.length === prevProps.messageListItems.length
```
Comment (WofWca) [inline src/renderer/components/message/MessageList.tsx:648]: I guess this will fix the particular bug, but, as you mentioned, I believe there is a bigger underlying problem with `memo`.
It was added here https://github.com/deltachat/deltachat-desktop/pull/2535 and was just a way to save memory, if I understand it correctly. After #2987 we are no longer using pages, but messageCache, which is id-addressed, so I suspect we no longer need this. Either way, the manual `memo` condition checking doesn't seem to be saving us too many re-renders - I tried to do a few things and it re-renders practically every time something happens. Perhaps need to just re-render it when any of the props change.

**Review by WofWca — CHANGES_REQUESTED** [review]

**Review by WofWca — COMMENTED** [review]
In short, I'd revisit / remove CHANGELOG changes. Otherwise it's merge-able.

File: CHANGELOG.md (hunk 1 above)
Comment (farooqkz) [reply CHANGELOG.md:2706]: I didn't. LazyVim did xD

File: src/renderer/components/message/MessageList.tsx (hunk 3 above)
Comment (farooqkz) [reply src/renderer/components/message/MessageList.tsx:648]: Let's already merge this PR and discuss about it later.

Comment (ralphtheninja) [reply src/renderer/components/message/MessageList.tsx:648]: Is the problem fixed if the `React.memo()` is removed altogether?

Comment (WofWca) [reply src/renderer/components/message/MessageList.tsx:648]: It should, though I did not check.
I'd keep `React.memo` and only remove `arePropsEqual`, but need to check whether it really is not needed anymore.

Comment (farooqkz) [reply src/renderer/components/message/MessageList.tsx]: Removed manual check

**Comment by nicodh** [comment]
I just tested this PR and it didn't fix the problem. The new message is still shown only in Chatlist but the message list stays empty...

**Comment by nicodh** [comment]
> > I just tested this PR and it didn't fix the problem. The new message is still shown only in Chatlist but the message list stays empty...
> 
> You sure. I just tested and it solves the problem. @Simon-Laux @WofWca What about you?

Tested again and still the new message is not shown if the chat was muted before. The chatlist shows the batch for a new message but messageList stays empty

**Comment by WofWca** [comment]
Indeed, it doesn't appear to fix the bug.
Though I tested it differently, instead of sending a message from a second device of the active account, I send it from the account with whom the chat is open on DC desktop.

**Comment by WofWca** [comment]
I discovered that when the bug happens, when a new message arrives, the `MessageList` re-renders, but `messageListItems` is always empty here

https://github.com/deltachat/deltachat-desktop/blob/9cb8c0d76ca680ef9086e7384c41868e1e3f41f9/src/renderer/components/message/MessageList.tsx#L115

So it doesn't loo like it has to do with `MessageListInner` being memoized.

**Comment by farooqkz** [comment]
> Indeed, it doesn't appear to fix the bug. Though I tested it differently, instead of sending a message from a second device of the active account, I send it from the account with whom the chat is open on DC desktop.

then it's a different bug?

**Comment by farooqkz** [comment]
@WofWca @nicodh I think we are talking about different bugs. This PR with its current state fixes the bug in the mentioned issue. However, I think maybe fix other related bugs in the same PR, too. Can you post exactly how do you produce the bugs, please?

**Comment by WofWca** [comment]
Indeed, those look like different bugs, and for some reason [I cannot reproduce the original bug even on `main` branch](https://github.com/deltachat/deltachat-desktop/issues/3991#issuecomment-2227288348) XD.

**Comment by Simon-Laux** [comment]
Superceded by #4037
